### PR TITLE
{chem}[foss/2018b,intel/2018b] SALMON v1.2.1

### PR DESCRIPTION
**Pull request by migueldiascosta**
(created using `eb --new-pr`)

requires https://github.com/easybuilders/easybuild-easyblocks/pull/1750

### CODE REVIEW

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/6725adb09f9409d7dfaba90708c1ae50 for a full test report.

File: easybuild/easyconfigs/s/SALMON/SALMON-1.2.1-intel-2018b.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+checksums = ['a5045149e49abe9dd9edefe00cd1508a1323081bc3d034632176b728effdbaeb']
+
+builddependencies = [('CMake', '3.12.1')]
+
+dependencies = [('libxc', '4.2.3')]
+
+separate_build_dir = True
+
+configure_cmd = "python ../SALMON-v.%(version)s/configure.py"
+configopts = "FC='$MPIF90' CC='$MPICC' FFLAGS='$FFLAGS' CFLAGS='$CFLAGS' "
```
Comment (boegel) [inline easybuild/easyconfigs/s/SALMON/SALMON-1.2.1-intel-2018b.eb]: Wait, this doesn't make sense... If you use single quotes around `$CFLAGS` & co, they won't get expanded by the shell (it'll be a literal `'$CFLAGS'` value).

So, why does this work?! :)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/e17be8da77d3e7932b23633314cde419 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2045.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5c69f23a86d90a02aab9e7def0bef3a0 for a full test report.

File: easybuild/easyconfigs/s/SALMON/SALMON-1.2.1-intel-2018b.eb (hunk 1 above)
Comment (migueldiascosta) [reply easybuild/easyconfigs/s/SALMON/SALMON-1.2.1-intel-2018b.eb]: you're right... but this are arguments to the `configure.py` script, which then does:

```python
define = ''
for k,v in dict.items():
  define = '{0} -D {1}={2}'.format(define, k, v)

env = ''
for var in args:
  (k, v) = var.split('=', 1)
  env = '{0} {1}="{2}"'.format(env, k, v)

### configuration                                                                                                  
comm = '{2} cmake {0} {1}'.format(define, SOURCE_DIR, env)
print('    $ %s' % comm)
os.system(comm)

```
so in this case I suppose it's actually better that it is not expanded here, only in the `cmake` command executed by the `configure.py` script (?)

Comment (boegel) [reply easybuild/easyconfigs/s/SALMON/SALMON-1.2.1-intel-2018b.eb]: Ah, that explains it... Thanks for the clarification! (also, ugh...)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2601.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2098883eab6e654cf4510b38c7412ce1 for a full test report.

**Comment by boegel** [comment]
Verified that correct compiler & options are being used by checking log after adding `buildopts = 'V=1 VERBOSE=1'` to the easyconfig, so good to go...

**Comment by boegel** [comment]
Going in, thanks @migueldiascosta!

**Pull request merged**